setup.py

```diff
@@ -19,14 +19,15 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 import os
 import re
+import sys
 import types
 
 import setuptools
 
-name = "hikari"
+if sys.version_info < (3, 8, 0):
+    raise RuntimeError("hikari requires Python 3.8.0 or greater")
```

Comment:
isn't this what python_requires is for

Reply:
Unfortunately `python_requires` its only an indicator. Pip enforces that indicator but if you run `python setup.py install`, it completely ignores it. This makes sure its enforced

Reply:
Well one just because it cannot be installed into an environment doesn't mean it can't be built in that environment and that check seems to cover a wider scope than intended. 
Also that behaviour kinda sounds like normal expected behaviour? It's pip which is strict about this stuff; you don't get stuff like strict requirement cross-checks when you install using setup.py either and probably shouldn't be using setup.py directly if you want a safe install (wouldn't ever have setup.cfg raising because of that stuff either).